Sources/TuistSigning/SigningCipher.swift

```diff
@@ -0,0 +1,151 @@
+import Basic
+import CryptoSwift
+import Foundation
+import TuistCore
+import TuistSupport
+
+enum SigningCipherError: FatalError, Equatable {
+    case failedToEncrypt
+    case failedToDecrypt(String)
+    case ivGenerationFailed(String)
+    case masterKeyNotFound(AbsolutePath)
+    case signingDirectoryNotFound(AbsolutePath)
+
+    var type: ErrorType { .abort }
```

Comment:
I'd implement all the cases. That way, if someone adds a new case in the future, they'll forced to think about the type.

Reply:
This just declares what the `ErrorType` of `SigningCipher` is, I suppose a switch does not make sense in this case.

Reply:
Right, but as a developers we need to think about what type of error we are throwing. If we don't use a switch clause and rather always return the same value, when I add a new value I won't be prompted to think about the type of error of the new case.

```swift
var type: ErrorType {
  switch self {
    ...
  } 
}
```

Reply:
Oh sorry, you're definitely right, I kind of did not realize it makes sense to have different `ErrorType` for individual error cases 👍